docs/developer/contributing/running-kibana-with-apm.asciidoc

```diff
@@ -0,0 +1,70 @@
+
+This type of data can help us identify unknown bottlenecks, spot when a performance regression may have been introduced, and inform how the performance of Kibana is changing between releases. Using APM allows us to be proactive in getting ahead of potential performance regressions before they are released.
+
+The easiest and recommended way of running Kibana with the APM agent locally is to use the solution provided by the https://github.com/elastic/apm-integration-testing[apm-integration-testing] repo. You’ll need https://www.docker.com/community-edition[Docker] and https://docs.docker.com/compose/install/[Docker Compose] to use the tool.
+
+[discrete]
+=== Quick start guide
+
+1. Clone https://github.com/elastic/apm-integration-testing[elastic/apm-integration-testing]
+2. Change into the apm-integration-testing repo:
+** `cd apm-integration-testing`
+3. Run {es} and the APM servers without Kibana using:
+** `./scripts/compose.py start master --no-kibana`
+4. Change into the {kib} repo:
+** `cd ../kibana`
+5. Change the elasticsearch credentials in your `kibana.yml` configuration file to match those needed by elasticsearch and the APM server (see the apm-integration-testing repo's https://github.com/elastic/apm-integration-testing#logging-in[README] for users provided to test different scenarios).
+6. Make sure that the APM agent is active and points to the local APM server by adding the following configuration settings to your `kibana.yml` configuration file:
++
+["source","shell"]
+----------
+elastic.apm.active: true
+elastic.apm.serverUrl: 'http://127.0.0.1:8200'
+# elastic.apm.secretToken: ... <-- might be required in prod/cloud
+# optional metrics to tune the performance of APM agents
+elastic.apm.centralConfig: false
+elastic.apm.breakdownMetrics: false
+elastic.apm.transactionSampleRate: 0.1
+elastic.apm.metricsInterval: '120s'
+----------
+
+7. Start Kibana with APM active using:
+** `yarn start`
+8. After Kibana starts up, navigate to the APM app, where you should see some transactions.
+
+image::images/apm_ui_transactions.png[]
+
+You can now continue doing what you want to in Kibana (e.g. install sample data sets, issue queries in dashboards, build new visualizations etc).
+Once you're finished, you can stop Kibana normally, then stop the {es} and APM servers in the apm-integration-testing clone with the following script:
+** `./scripts/compose.py stop`
+
+If you don't want to add these settings to your `config/kibana.yml` file directly, you can add them to a config file under `config/apm.dev.js`:
+
+Example `config/apm.dev.js` file:
+
+[source,js]
+----
+module.exports = {
+  active: true,
+  serverUrl: 'http://127.0.0.1:8200', // use `http://localhost:8200` as an alternative
```

Comment:
Self review: using `http://127.0.0.1:8200` doesn't work for me, I'm forced to use `localhost`. The apm-agent-node team is trying to figure out if something about the Node.js agent (in a recent release) is preventing it from properly parsing "http://127.0.0.1:8200" when "http://localhost:8200" works OR if the problem's communicating with the APM Server and it can't receive traffic at 127.0.0.1 but can at localhost.